fastplotlib/widgets/histogram_lut.py

```diff
@@ -205,27 +239,39 @@ def _linear_region_handler(self, ev):
     def _image_cmap_handler(self, ev):
         setattr(self, ev.type, ev.info["value"])
 
+    @property
+    def cmap(self) -> str:
+        return self._cmap
+
+    @cmap.setter
+    def cmap(self, name: str):
+        if self._colorbar is None:
+            return
+
+        with pause_events(self.image_graphic):
+            self.image_graphic.cmap = name
+
+            self._cmap = name
+            self._colorbar.cmap = name
+
     @property
     def vmin(self) -> float:
         return self._vmin
 
     @vmin.setter
     def vmin(self, value: float):
-        self.image_graphic.block_events = True
-        self._linear_region_selector.block_events = True
-
-        # must use world coordinate values directly from selection()
-        # otherwise the linear region bounds jump to the closest bin edges
-        self._linear_region_selector.selection = (
-            value * self._scale_factor,
-            self._linear_region_selector.selection[1],
-        )
-        self.image_graphic.vmin = value
-
-        self.image_graphic.block_events = False
-        self._linear_region_selector.block_events = False
+        with pause_events(self.image_graphic, self._linear_region_selector):
+            # must use world coordinate values directly from selection()
+            # otherwise the linear region bounds jump to the closest bin edges
+            self._linear_region_selector.selection = (
+                value * self._scale_factor,
+                self._linear_region_selector.selection[1],
+            )
+            self.image_graphic.vmin = value
```

Comment:
@clewis7 you might also find this useful, just pass any number of graphics to `pause_events()` and the events are blocked for those graphics while inside the `with` block :smile: